packages/atoms/availability-list/index.tsx

```diff
@@ -14,13 +14,13 @@ export type Schedule = {
     id: number;
     startTime: Date;
     endTime: Date;
-    userId?: number;
-    eventTypeId?: number;
-    date?: Date;
+    userId: number | null;
```

Comment:
@supalarry the availability object in type `Schedule` is expected to have a value and that's why setting it to optional gives a type error since it will be `value | undefined` that way instead of `value | null`

Reply:
Interesting, I checked `Availability` table locally and indeed `userId` can be `null` - i also thought that it must always be present.

Reply:
> Interesting, I checked `Availability` table locally and indeed `userId` can be `null` - i also thought that it must always be present.

I tried passing the schedule down to a method inside a function and thats when I started getting type errors, I think maybe the components or methods down to which it is being passed expect its value to never be undefined

Reply:
In `schema.prisma` Availability userId type is `Int?` so if userId is not passed it will be null in database

Reply:
Yeah that might be the case what I'm trying to say is the methods which take in `schedule.availability` always expect a value (null included) and as a result I get type errors when trying to make `availability` optional

Reply:
All good - understood!